src/Symfony/Component/Encryption/Phpseclib/PhpseclibEncryption.php

```diff
@@ -0,0 +1,175 @@
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ *
+ * @experimental in 5.3
+ */
+class PhpseclibEncryption implements EncryptionInterface
+{
+    public function __construct()
+    {
+        if (!class_exists(RSA::class)) {
+            throw new \LogicException('You cannot use "Symfony\Component\Security\Core\Encryption\PhpseclibEncryption" as the "phpseclib/phpseclib:2.x" package is not installed. Try running "composer require phpseclib/phpseclib:^2".');
+        }
+    }
+
+    public function generateKey(string $secret = null): KeyInterface
+    {
+        $rsa = new RSA();
+        $key = $rsa->createKey();
+
+        if ($key['partialkey']) {
+            throw new EncryptionException('Failed to generate RSA keypair.');
+        }
+
+        if (null === $secret) {
+            $secret = random_bytes(32);
+        }
+
+        return PhpseclibKey::create($secret, $key['privatekey'], $key['publickey']);
+    }
+
+    public function encrypt(string $message, KeyInterface $myKey): string
+    {
+        if (!$myKey instanceof PhpseclibKey) {
+            throw new InvalidKeyException();
+        }
+
+        set_error_handler(__CLASS__.'::throwError');
+
+        try {
+            $aes = new AES(AES::MODE_CBC);
+            $aes->setPassword($myKey->getSecret());
+            $aes->setIV($nonce = Random::string($aes->getBlockLength() >> 3));
+
+            return Ciphertext::create('AES-CBC', $aes->encrypt($message), $nonce)->getString();
+        } catch (\ErrorException $exception) {
+            throw new EncryptionException(null, $exception);
+        } finally {
+            restore_error_handler();
+        }
+    }
+
+    public function encryptFor(string $message, KeyInterface $recipientKey): string
+    {
+        if (!$recipientKey instanceof PhpseclibKey) {
+            throw new InvalidKeyException();
+        }
+
+        try {
+            $rsa = new RSA();
+            $rsa->loadKey($recipientKey->getPublicKey());
+            $rsa->setEncryptionMode(RSA::ENCRYPTION_OAEP);
+
+            return Ciphertext::create('RSA-OAEP', $rsa->encrypt($message), random_bytes(8))->getString();
+        } catch (\ErrorException $exception) {
+            throw new EncryptionException(null, $exception);
+        } finally {
+            restore_error_handler();
+        }
+    }
+
+    public function encryptForAndSign(string $message, KeyInterface $keypair): string
+    {
+        if (!$keypair instanceof PhpseclibKey) {
+            throw new InvalidKeyException();
+        }
+        try {
+            $rsa = new RSA();
+            $rsa->loadKey($keypair->getPublicKey());
+            $rsa->setEncryptionMode(RSA::ENCRYPTION_OAEP);
+            $ciphertext = $rsa->encrypt($message);
+
+            // Load private key after encryption
+            $rsa->loadKey($keypair->getPublicKey());
+            $rsa->setSignatureMode(RSA::SIGNATURE_PSS);
+            $headers['signature'] = base64_encode($rsa->sign($ciphertext));
```

Comment:
I've never seen this style?
```php
$headers = ['signature' => base64_encode($rsa->sign($ciphertext))];
```

Reply:
What do you mean?

Reply:
I mean, i've never seen an array defined the way you have.

Reply:
```php
$headers = [];
$headers = ['foo'=>'bar'];
$headers = ['foo' => json_encode('bar')];
```

Sure I could write something like this, but that is an extra line. 

```php 
$headers = [];
$headers['signature'] = base64_encode($rsa->sign($ciphertext));
```